src/frontend/src/common.tsx

```diff
@@ -542,12 +564,25 @@ export const UserLink = ({
     id,
     classNameArg,
     profile,
+    name,
 }: {
     id: UserId;
     classNameArg?: string;
     profile?: boolean;
+    name?: string;
 }) => {
-    const userName = window.backendCache.users[id];
```

Comment:
That's an interesting idea, but what I'd do first is to measure the hit/miss ratio: I am afraid that in many cases where we need to load say 20 user names (e.g. by opening the post info section with reactions, watchers, tippers and so on), we still would have a small number of misses because we never loaded posts from those users and would still need to do a query. But in that case the latency would be already not zero...

Reply:
It would be useful when the user returns to the same posts / comments / reactions multiple times.